File: Python/ByRound/0883/0883_M_Quadcopter_Competition.py
# Problem: CF 883 M - Quadcopter Competition
# https://codeforces.com/contest/883/problem/M

"""
Code purpose: 
    This code calculates the minimal path length for a quadcopter to go around a flag and return to its starting point.
    The quadcopter moves on a grid and can only move in four directions (up, down, left, right) by one unit per step.
    
    The problem is about finding the shortest closed path that strictly contains the flag point (x2, y2) inside the cycle.
    We are given the starting point (x1, y1) and the flag point (x2, y2).
    
    The key insight is to compute the Manhattan distance between the start and flag points, then determine
    how to construct the minimal cycle based on whether the flag is aligned vertically or horizontally with the start.
    
Algorithms/Techniques:
    - Manhattan Distance Calculation
    - Special Case Handling: When start and flag are aligned (same x or same y coordinates)
    
Time Complexity: O(1) - constant time operations
Space Complexity: O(1) - only using a few variables
"""

a, b = map(int, input().split())  # Read starting point coordinates
c, d = map(int, input().split())  # Read flag coordinates

# Check if the start and flag are aligned (same x or same y)
if a == c or b == d:
    # If aligned, the minimal path adds 6 extra steps to go around the flag
    print(2 * abs(a - c) + 2 * abs(b - d) + 6)
else:
    # If not aligned, the minimal path adds 4 extra steps
    print(2 * abs(a - c) + 2 * abs(b - d) + 4)


# https://github.com/VaHiX/CodeForces/

File: CPP/ByTier/M/0883_M_Quadcopter_Competition.cpp
// Problem: CF 883 M - Quadcopter Competition
// https://codeforces.com/contest/883/problem/M

/*
Code Purpose:
This program calculates the minimal length of a closed path a quadcopter must take to go from a starting point to a flag and return to the starting point. The path is constrained to move only vertically or horizontally (parallel to the axes), one unit per step. The flag must be strictly inside the path.

Algorithm:
The key insight is that the minimal closed path forming a cycle that contains the flag is a rectangle around the flag with an additional buffer to ensure the flag is strictly inside. The path is calculated based on Manhattan distances from the start to the flag and then expanding the path by at least one unit in all directions to meet the "strictly inside" requirement.

Time Complexity: O(1) - The solution performs a constant number of operations.
Space Complexity: O(1) - Only a constant amount of space is used.
*/

#import <bits/stdc++.h>
using namespace std;
int a, b, c, d;
int main() {
  cin >> a >> b >> c >> d; // Read starting point (a,b) and flag point (c,d)
  a = abs(a - c), b = abs(b - d); // Calculate Manhattan distance in x and y directions
  cout << 2 * (a + !a + b + !b + 2); // Output the minimal path length using derived formula
}


// https://github.com/VaHiX/CodeForces/